src/app/site/cart/components/summary/summary.component.html

```diff
@@ -27,40 +27,25 @@ <h2 style="white-space: nowrap">عضویت رایگان</h2>
   </div>
   <hr class="hr-style">
   <!-- coupon -->
-  <div class="rows">
-    <div fxLayout="row" fxLayoutAlign="center center" fxLayoutGap="0px">  
-        <span class="coupon-name">کوپن تخفیف</span>
-        <input class="coupon-input" matInput type="text">
+  <div class="rows coupon-rows">
+    <div fxLayout="row">
+      <div fxFlex="40" class="coupon-name">کوپن تخفیف</div>
+      <div fxFlex="60" style="padding: 1px 0 0 2px">
+        <input matInput *ngIf="!used_coupon_code" class="coupon-input" type="text" [(ngModel)]="coupon_code" title="coupon code"/>
+        <input matInput *ngIf="!!used_coupon_code" class="coupon-input" type="text" [value]="used_coupon_code" title="coupon code"
+               readonly/>
+      </div>
     </div>
-    <div fxLayout="row" fxLayoutAlign="center center" fxLayoutGap="0px">
-        <button class="coupon-btn" mat-raised-button>اعمال</button>
+    <div>
+      <button class="coupon-btn" mat-button *ngIf="!used_coupon_code" (click)="applyCoupon()">اعمال</button>
+      <button class="coupon-btn" mat-button *ngIf="used_coupon_code" (click)="applyCoupon()">حذف</button>

```

Comment:
please remove this tow (click)="applyCoupon()", and add it to div, to avoid from redundancy and better user experience